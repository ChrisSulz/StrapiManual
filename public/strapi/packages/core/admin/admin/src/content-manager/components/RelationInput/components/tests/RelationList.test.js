import React from 'react';

import { lightTheme, ThemeProvider } from '@strapi/design-system';
import { render } from '@testing-library/react';
import { DndProvider } from 'react-dnd';
import { HTML5Backend } from 'react-dnd-html5-backend';

import { RelationItem } from '../RelationItem';
import { RelationList } from '../RelationList';

const setup = ({ endAction }) =>
  render(
    <ThemeProvider theme={lightTheme}>
      <DndProvider backend={HTML5Backend}>
        <RelationList>
          <RelationItem
            ariaDescribedBy="test"
            iconButtonAriaLabel="Drag"
            index={0}
            id={0}
            name="test_field"
            endAction={endAction}
          >
            First relation
          </RelationItem>
          <RelationItem
            ariaDescribedBy="test"
            iconButtonAriaLabel="Drag"
            index={1}
            id={1}
            name="test_field"
            endAction={endAction}
          >
            Second relation
          </RelationItem>
        </RelationList>
      </DndProvider>
    </ThemeProvider>
  );

describe('Content-Manager || RelationInput || RelationList', () => {
  it('should render and match snapshot', () => {
    const { container } = setup({});

    expect(container).toMatchSnapshot();
  });
});
