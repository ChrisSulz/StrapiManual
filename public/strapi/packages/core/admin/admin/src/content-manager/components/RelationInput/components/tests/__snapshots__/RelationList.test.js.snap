// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content-Manager || RelationInput || RelationList should render and match snapshot 1`] = `
.c7 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c1 {
  cursor: default;
}

.c2 {
  background: #ffffff;
  padding-top: 8px;
  padding-right: 16px;
  padding-bottom: 8px;
  padding-left: 16px;
  border-radius: 4px;
  border-color: #dcdce4;
  border: 1px solid #dcdce4;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 4px;
}

.c5 {
  width: 100%;
  min-width: 0;
}

.c5 > div[role='button'] {
  cursor: all-scroll;
}

.c6 {
  width: 100%;
  min-width: 0;
}

.c0 {
  position: relative;
  overflow: hidden;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0:before,
.c0:after {
  position: absolute;
  width: 100%;
  height: 4px;
  z-index: 1;
}

.c0:before {
  content: '';
  background: linear-gradient(rgba(3,3,5,0.2) 0%,rgba(0,0,0,0) 100%);
  top: 0;
  opacity: 0;
  -webkit-transition: opacity 0.2s ease-in-out;
  transition: opacity 0.2s ease-in-out;
}

.c0:after {
  content: '';
  background: linear-gradient(0deg,rgba(3,3,5,0.2) 0%,rgba(0,0,0,0) 100%);
  bottom: 0;
  opacity: 0;
  -webkit-transition: opacity 0.2s ease-in-out;
  transition: opacity 0.2s ease-in-out;
}

<div>
  <div
    class="c0"
  >
    <li
      aria-describedby="test"
      class="c1"
    >
      <div
        class="c2 c3"
        data-handler-id="T0"
      >
        <div
          class="c4 c5"
        >
          <div
            class="c3 c6"
          >
            First relation
          </div>
        </div>
      </div>
    </li>
    <li
      aria-describedby="test"
      class="c1"
    >
      <div
        class="c2 c3"
        data-handler-id="T2"
      >
        <div
          class="c4 c5"
        >
          <div
            class="c3 c6"
          >
            Second relation
          </div>
        </div>
      </div>
    </li>
  </div>
  <div
    class="c7"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
